constants.go

```diff
@@ -102,5 +102,5 @@ const (
 
 var (
 	// DefaultSamplingServerURL is the default url to fetch sampling config from, via http
-	DefaultSamplingServerURL = fmt.Sprintf("http://localhost:%d/sampling", DefaultSamplingServerPort)
+	DefaultSamplingServerURL = fmt.Sprintf("http://127.0.0.1:%d/sampling", DefaultSamplingServerPort)
```

Comment:
why this change?

Reply:
It's not a big reason.
I referred to [csharp change](https://github.com/jaegertracing/jaeger-client-csharp/pull/165), 127.0.0.1 was used in both readme.md and code. 
```
JAEGER_SAMPLING_ENDPOINT | The url for the remote sampling conf when 
using sampler type remote. Default is http://127.0.0.1:5778/sampling
```

If you want, we can stay current code since localhost is 127.0.0.1 (except /etc/hosts lookup.)